src/cmap/auth/mongodb_oidc.ts

```diff
@@ -95,19 +98,32 @@ export class MongoDBOIDC extends AuthProvider {
     handshakeDoc: HandshakeDocument,
     authContext: AuthContext
   ): Promise<HandshakeDocument> {
-    const { credentials } = authContext;
-
-    if (!credentials) {
-      throw new MongoMissingCredentialsError('AuthContext must provide credentials.');
+    const credentials = getCredentials(authContext);
+    const { connection } = authContext;
+    const mechanismProperties = credentials.mechanismProperties;
+    const allowedHosts = mechanismProperties.ALLOWED_HOSTS || DEFAULT_ALLOWED_HOSTS;
+    const providerName = mechanismProperties.PROVIDER_NAME;
+    // Cloud provider automatic auth does not validate allowed hosts.
+    if (!providerName && !hostMatchesWildcards(connection.address, allowedHosts)) {
+      throw new MongoInvalidArgumentError('Host does not match provided ALLOWED_HOSTS values');
```

Comment:
Does it make sense to include the hostname in the error message? That feels like it could improve debugging quite a bit.

Reply:
I've updated to include the hostname and ALLOWED_HOSTS in the error message.